### PR TITLE
Doing some refactoring + catching up the pg_proxy example w/the duckdb one

### PR DESCRIPTION
**Pull request by jwills**
Will modularize this stuff under `buenavista/playground` this week

### CODE REVIEW

File: buenavista/types.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        if not converter:
+            self.converter = lambda x: str(x)
+        else:
+            self.converter = converter
+        self._instances[oid] = self
+
+    @classmethod
+    def find_by_oid(cls, oid: int) -> Optional["PGType"]:
+        t = cls._instances.get(oid)
+        return PGTypes.UNKNOWN if not t else t
```
Comment (Mause) [inline buenavista/types.py:23]:
```suggestion
        return cls._instances.get(oid, PGTypes.UNKNOWN)
```

File: buenavista/types.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    _instances = {}
+
+    def __init__(self, oid, converter=None):
+        if oid in self._instances:
+            raise ValueError(f"Duplicate OID {oid} declared for PGType")
+        self.oid = oid
+        if not converter:
+            self.converter = lambda x: str(x)
+        else:
+            self.converter = converter
```
Comment (Mause) [inline buenavista/types.py:17]:
```suggestion
        self.converter = converter or str
```

File: examples/pg_proxy.py [hunk 3]
```diff
@@ -25,68 +31,53 @@ def t(row):
...
-        self.db.autocommit = True
-
-    def _cursor(self):
-        return self.db.cursor()
-
-    def parameters(self) -> Dict[str, str]:
-        return {"server_version": "BV.pg8000.1", "client_encoding": "UTF8"}
+class PGAdapterHandle(AdapterHandle):
+    def __init__(self, cursor):
+        super().__init__(cursor)
```
Comment (Mause) [inline examples/pg_proxy.py:36]:
```suggestion
```

Don't need to redefine the init if it's identical

**Review by Mause — COMMENTED** [review]
Some small suggestions 😁

File: examples/pg_proxy.py (hunk 3 above)
Comment (jwills) [reply examples/pg_proxy.py:36]: Ack thanks! Still not very good at Python 😉

**Pull request merged**